lib/src/data/user_profile.dart

```diff
@@ -0,0 +1,75 @@
+import "course.dart";
+import "utils.dart";
+
+/// Represents information about a normal user.
+class UserProfile {
+  /// The user's name.
+  final String name;
+
+  /// The user's current courses.
+  final List<Course> currentCourses;
+
+  /// The user's past courses.
+  final List<Course> pastCourses;
+
+  /// The user's current school year
+  final SchoolYear year;
+
+  /// Creates a seller's profile.
+  const UserProfile({
+    required this.name,
+    required this.currentCourses,
+    required this.pastCourses,
+    required this.year,
+  });
+
+  /// Creates a new UserProfile object from a JSON object.
+  UserProfile.fromJson(Json json) :
+    name = json["name"],
+    currentCourses = [
+      for (final courseJson in json["currentCourses"])
+        Course.fromJson(courseJson),
+    ],
+    pastCourses = [
+      for (final courseJson in json["currentCourses"])
+        Course.fromJson(courseJson),
+    ],
+    year = SchoolYear.fromJson(json["year"]);
+
+  /// Convert this UserProfile to its JSON representation
+  Json toJson() => {
+    "name": name,
+    "currentCourses": [
+      for (final currentCourse in currentCourses)
+        currentCourse.toJson(),
+    ],
+    "pastCourses": [
+      for (final pastCourse in pastCourses)
+        pastCourse.toJson(),
+    ],
+    "year": year.toJson(),
+  };
+}
+
+/// A year in school
+enum SchoolYear {
+  freshman(id: 0),
+  sophomore(id: 1),
+  junior(id: 2),
+  senior(id: 3),
+  graduate(id: 4),
+  faculty(id: 5);
```

Comment:
These don't need explicit IDs because Dart already gives `enumValue.index`. As for JSON logic, if you don't want this enum's order to become a breaking change, you can use the index or name of the value: 

```dart
"year": year.index,
year = SchoolYear.values[json["year"]];
// or
"year": year.name,
year = SchoolYear.values.byName(json["year"]);
```

In more sensitive APIs though. having an explicit ID that devs know not to change can be a good thing though.

Reply:
For future reference, I went with by index.